Vic2ToHoI4/Source/HOI4World/States/DefaultStatesImporter.cpp

```diff
@@ -0,0 +1,25 @@
+#include "DefaultStatesImporter.h"
+#include "OSCompatibilityLayer.h"
+
+
+
+HoI4::DefaultStatesImporter::DefaultStatesImporter()
+{
+	registerKeyword("state", [this](std::istream& theStream) {
+		defaultStates_.insert(std::make_pair(num_, DefaultState(theStream)));
+	});
+}
+
+
+
+std::map<int, HoI4::DefaultState> HoI4::DefaultStatesImporter::ImportDefaultStates(const std::string& hoi4Path)
+{
+	const std::string path = hoi4Path + "/history/states";
+	for (const auto& stateFile: commonItems::GetAllFilesInFolder(path))
+	{
+		num_ = std::stoi(stateFile.substr(0, stateFile.find_first_of('-')));
+		parseFile(path + '/' + stateFile);
```

Comment:
I suggest looking for an equivalent of C#'s Path.Combine function. Takes care of adding slashes and prevents adding a slash to a path already ending with a slash.